src/django_components/component_registry.py

```diff
@@ -223,10 +225,19 @@ def __init__(
         self._registry: Dict[str, ComponentRegistryEntry] = {}  # component name -> component_entry mapping
         self._tags: Dict[str, Set[str]] = {}  # tag -> list[component names]
         self._library = library
-        self._settings_input = settings
-        self._settings: Optional[Callable[[], InternalRegistrySettings]] = None
+        self._settings = settings
 
-        all_registries.append(self)
+        ALL_REGISTRIES.append(cached_ref(self))
+
+    def __del__(self) -> None:
```

Comment:
I also added some lifecycle mgmt hooks to ComponentRegistries:
- When a registry is being deleted, we now first unregister all components, so that the associated `Library` instance is properly cleaned up.
- When a Component class is deleted, then it will be automatically unregistered from all Registries it was registered in ([see here](https://github.com/django-components/django-components/pull/1008/files#diff-a5dcf86b1b1bd690092e8ed60318835a1ce9ed2c62befcc05d43dfd4459393b6R329)). Again to ensure that the associated `Library` instance is cleaned up properly.